Public/Src/Tools/Execution.Analyzer/Analyzers.Core/XLGPlusPlus/XldbProtobufExtensions.cs

```diff
@@ -558,5 +548,479 @@ public static Xldb.Fingerprint ToFingerprint(this Fingerprint fingerprint)
+        {
+            var xldbSealDirectory = new Xldb.SealDirectory
+            {
+                ParentPipInfo = parentPip,
+                Kind = (SealDirectoryKind)pip.Kind,
+                DirectoryRoot = pip.DirectoryRoot.ToAbsolutePath(pathTable),
+                IsComposite = pip.IsComposite,
+                Scrub = pip.Scrub,
+                IsInitialzed = pip.IsInitialized,
+                Directory = pip.Directory.ToDirectoryArtifact(pathTable),
+                IsSealSourceDirectory = pip.IsSealSourceDirectory,
+                Provenance = pip.Provenance.ToPipProvenance(),
+                PipType = (PipType)pip.PipType
+
+            };
+
+            xldbSealDirectory.Patterns.AddRange(pip.Patterns.Select(key => key.ToString()));
+            xldbSealDirectory.Contents.AddRange(pip.Contents.Select(file => file.ToFileArtifact(pathTable)));
+            xldbSealDirectory.ComposedDirectories.AddRange(pip.ComposedDirectories.Select(dir => dir.ToDirectoryArtifact(pathTable)));
+
+            if (pip.Tags.IsValid)
+            {
+                xldbSealDirectory.Tags.AddRange(pip.Tags.Select(key => key.ToString()));
+            }
+
+            return xldbSealDirectory;
+        }
+
+        /// <nodoc />
+        public static Xldb.CopyFile ToCopyFile(this CopyFile pip, PathTable pathTable, Xldb.Pip parentPip)
+        {
+            var xldbCopyFile = new Xldb.CopyFile
+            {
+                ParentPipInfo = parentPip,
+                Source = pip.Source.ToFileArtifact(pathTable),
+                Destination = pip.Destination.ToFileArtifact(pathTable),
+                OutputsMustRemainWritable = pip.OutputsMustRemainWritable,
+                Provenance = pip.Provenance.ToPipProvenance(),
+                PipType = (PipType)pip.PipType
+            };
+
+            if (pip.Tags.IsValid)
+            {
+                xldbCopyFile.Tags.AddRange(pip.Tags.Select(key => key.ToString()));
+            }
+
+            return xldbCopyFile;
+        }
+
+        /// <nodoc />
+        public static Xldb.WriteFile ToWriteFile(this WriteFile pip, PathTable pathTable, Xldb.Pip parentPip)
+        {
+            var xldbWriteFile = new Xldb.WriteFile
+            {
+                ParentPipInfo = parentPip,
+                Destination = pip.Destination.ToFileArtifact(pathTable),
+                Contents = pip.Contents.ToPipData(),
+                Encoding = (WriteFileEncoding)pip.Encoding,
+                Provenance = pip.Provenance.ToPipProvenance(),
+                PipType = (PipType)pip.PipType
+            };
+
+            if (pip.Tags.IsValid)
+            {
+                xldbWriteFile.Tags.AddRange(pip.Tags.Select(key => key.ToString()));
+            }
+
+            return xldbWriteFile;
+        }
+
+        /// <nodoc />
+        public static ProcessPip ToProcessPip(this Process pip, PathTable pathTable, Xldb.Pip parentPip)

```

Comment:
>ToProcessPip [](start = 33, length = 12)

General comment: You did a great job of faithfully of replicating every piece of data on the Process pip. This is great for completeness but I'm wondering if we should be more selective about exposing a smaller subset of properties for the sake of making it easier to keep the schema consistent.

An incomplete list of things I'm not sure we should lead with:
Metapips (Spec, module, value, HashSourceFile)
Priority
Warning/Error regex
ProducedPathIndependentOutputs
Container related data
...
Basically things that are more about the inner workings of the scheduler or obscure features


This would be a good thing to discuss with Oleksii. We can always add things later.

Reply:
Based on offline chat we have removed some things from the proto structs

---
In reply to: [314456761](https://github.com/microsoft/BuildXL/pull/724#discussion_r314456761) [](ancestors = 314456761)